explainers/signed-exchange-subresource-subtitution-explainer.md

```diff
@@ -0,0 +1,315 @@
+failure.
+
+## Multiple subresource signed exchanges
+If there are multiple matching subresource signed exchanges (example:
+script.js.sxg and image.jpg.sxg), the UA must check that there is no error in
+the all signed exchanges (eg: sig matching, URL matching, Merkle Integrity
+error) before processing the content of the signed exchanges. This is intended
+to prevent the distributor from encoding a tracking ID into the set of
+subresources it prefetches. this means that the UA can use the subresource
+signed exchanges only when they are defined in the header.
+
+## Can’t we have a global SignedExchangeCache?
+If we can have a global SignedExchangeCache, we can use the all signed exchanges
+which have previously prefetched, even if they are not prefetched by the
+referrer page. This can improve the performance. But this introduces privacy
+issues such as
+[Timing Leaks](https://hillbrad.github.io/sri-addressable-caching/sri-addressable-caching.html#cache_timing)
+and [Deterministic History Leaks](https://hillbrad.github.io/sri-addressable-caching/sri-addressable-caching.html#history_leaks)
+described in [the document about Subresource Integrity Addressable Caching](https://hillbrad.github.io/sri-addressable-caching/sri-addressable-caching.html).
+
+To avoid these privacy issues, we introduced the limitation that we can only use
+the signed exchanges which were prefetched by the referrer page.
+
+## Can’t we merge allowed-alt-sxg to preload header?
+If we can declare the header-integrity value in the existing preload link HTTP
+header, we don’t need to introduce a new "allowed-alt-sxg" link HTTP header. But
+it becomes complicated when supporting the
+[imagesrcset attribute of preload link](https://github.com/w3c/preload/issues/120).
+
+For Example:
+```
+Link: <https://publisher.example/wide.jpg>;
+      rel="preload";
+      as="image";
+      imagesrcset="https://publisher.example/wide.jpg 640w,
+                   https://publisher.example/narrow.jpg 320w";
+      imagesizes="(max-width: 640px) 100vw, 640px"
+```
+In this case, we want to declare that both wide.jpg and narrow.jpg can be loaded
+from signed exchanges and the header-integrity is sha256-XXX and sha256-YYY. But
+how to express in this preload link header?
+
+In our proposal, we can have two allowed-alt-sxg link headers.
+```
+Link: <https://publisher.example/wide.jpg>;
+      rel="allowed-alt-sxg";
+      header-integrity="sha256-XXX"
+Link: <https://publisher.example/narrow.jpg>;
+      rel="allowed-alt-sxg";
+      header-integrity="sha256-YYY"
+```
+
+## Lifetime of the entry in SignedExchangeCache
+The UA must check both the
+[signature expire time](https://wicg.github.io/webpackage/loading.html#exchange-signature-expiration-time)
+of the signed exchange and Cache-Control header of the outer response. The UA
+may discard the entry if it is expired.
+
+## Content negotiation using Variants and Variant-Key
```

Comment:
Only include this section in one of the explainers.

Reply:
Merged.